lib/core/network/httpRouter/index.js

```diff
@@ -156,6 +155,21 @@ class Router {
       // Set Headers if not present
       routeHandler.request.response.setHeaders(this.defaultHeaders, true);
 
+      
+      if (message.headers && message.headers.origin) {
```

Comment:
Seems like duplicated code in **L158** & **L186** -
Is the intention to have explicit duplicated code or could this be abstracted?